src/systemctl/systemctl-list-unit-files.c

```diff
@@ -174,7 +170,6 @@ int verb_list_unit_files(int argc, char *argv[], void *userdata) {
                 }
```

Comment:
Hmm.
```C
                        units[c++] = *u;
                        free(u);
```
This moves ownership of contents from `u` to `units`. After that, `hashmap_free(h)` was used which would not touch the values in the hashmap. But after your patch, the hashmap is freed using `unit_file_list_hash_ops`, which would try to deallocate the contents again. I think.

Reply:
Oops. Fixed.